### ISSUE TITLE
Desafio do capítulo 17 -> funciona assim mesmo?

### ISSUE THREAD

**Issue by ghost** [andrelncampos/LivroCSharp#5]
Infelizmente não funcionou assim ou eu pelo menos consigo a fazê-lo funcionar. 

Não entendo bem a solução: porque o código da página @page "/desafio17" ainda contem nas linhas 76-84 essa coisa if com o filtro? E porque ainda existe o private string Filtro e o méthodo LimparFiltro() na parte do @code?

**Comment by andrelncampos** [comment]
Daniela, amanhã eu te respondo, ok? Vou avaliar direitinho e te responder.

Em seg, 25 de abr de 2022 16:43, DanielaSchindler ***@***.***>
escreveu:

**Comment by andrelncampos** [comment]
Oi Daniela, primeiro quero agradecer por compartilhar sua dúvida. Vou tentar responder.

Bem, relembrando, em uma aplicação Blazor nós temos duas partes: 1) HTML e 2) Código puro. Só que no Blazor nós podemos programar também na parte HTML, usando um @ e em seguida o comando, como em @if, @for, etc.

Neste desafio, nós temos uma propriedade ProdutoCadastro, que é uma instância de um produto apenas e é utilizado para fazer o binding com os inputs na tela de cadastro. Também temos a propriedade Filtro, que é uma string simples, onde podemos colocar o nome de um produto ou parte dele.

Nas linhas 73 a 104, ainda na parte HTML, eu programo a exibição da lista do que já tem cadastrado. Mas, onde está esta lista? Ele retorna de um método chamado ProdutosLista() que está na classe Produtos, na pasta Data.

Pois bem, a exibição do conteúdo se dá dentro de um foreach a partir da linha 73. Mas, onde entra o filtro?

Na linha 76 eu verifico primeiro se tem alguma informação dentro da propriedade Filtro. Se tem, então eu verifico na linha 79 se esta informação está continha na linha atual que estou exibindo da lista de produtos. Se for menor do que zero a resposta do IndexOf, então não contém. Se não contém, não deve ser exibido. Se não deve ser exibido, eu interrompo o loop com o continue, indo para o próximo produto sem exibir o corrente.

Assim, quando há alguma informação em Filtro, então eu consigo limitar a exibição apenas dos produtos que contém a esta informação.

Espero ter ajudado. Se ainda restarem dúvidas, compartilhe comigo, Ok?

**Comment by ghost** [comment]
Yeah! Agora funcionou! E eu entendi. O componente razor precisa dessas linhas para saber onde e o que exibir. Bacana!! Foi uma excelente  resposta!! Muito obrigada pela resposta rapidíssima e pela paciência!! Ajudou muito!!
